### PR TITLE
JedisCluster should support setting client name.

### PR DESCRIPTION
**Pull request by jiafu1115**
[Motivation]

JedisCluster should support setting client name similar with other types
of Jedis.

[Modifications]

(1) add private member: clientName in JedisClusterInfoCache;
(2) add new constructor with clientName for JedisCluster;

[Result]

JedisCluster support setting client name.

### CODE REVIEW

**Comment by jiafu1115** [comment]
@marcosnils @HeartSaVioR  we can solve this PR at first due to it is easier to change with low risk. Then we can continue focus on #1384

**Comment by marcosnils** [comment]
LGTM, we'll merge once another contributor approves.

**Comment by jiafu1115** [comment]
@marcosnils Thanks, and one question for it: why added this to 2.8.3 instead of 2.9.0. What's the difference between them.

**Comment by jiafu1115** [comment]
@marcosnils Thanks, and Let's wait @HeartSaVioR

**Comment by marcosnils** [comment]
> why added this to 2.8.3 instead of 2.9.0. What's the difference between them.

@jiafu1115 we use semver to tag our releases http://semver.org/. The milestone we assign to the issues it's the next version of Jedis that will have this change. We'll take care of merging to all the corresponding branches once it's approved and merged into master.

**Comment by jiafu1115** [comment]
@marcosnils OK， get it , so you will do your self. 
BTW:  the test build seems not stable. so I will ignore the test for this PR.
build 1452 - xetorthiojedis - travis ci - 360 8 1

**Comment by marcosnils** [comment]
@jiafu1115 tests are green here.

File: src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java [hunk 1]
```diff
@@ -29,12 +34,15 @@ public Jedis getConnectionFromNode(HostAndPort node) {
...
   }
 
-  private void initializeSlotsCache(Set<HostAndPort> startNodes, GenericObjectPoolConfig poolConfig, String password) {
+  private void initializeSlotsCache(Set<HostAndPort> startNodes, GenericObjectPoolConfig poolConfig, String password, String clientName) {
     for (HostAndPort hostAndPort : startNodes) {
       Jedis jedis = new Jedis(hostAndPort.getHost(), hostAndPort.getPort());
       if (password != null) {
         jedis.auth(password);
       }
+      if (clientName != null) {
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java]: This instance is for discovery and will be closed soon. Do we still want to apply client setname here?

**Comment by jiafu1115** [comment]
@HeartSaVioR.  If someday discovery command hang.we can know which app hang through client name. So setting client name for temp connection is to keep same style and provide possible help for trouble shooting. The important thing is it isnot bad to set it.WDYT

**Comment by HeartSaVioR** [comment]
@jiafu1115 Makes sense. LGTM. Thanks for the great work.

**Comment by HeartSaVioR** [comment]
Merged into master, 2.10, 2.9 branches.

**Comment by marcosnils** [comment]
thx @HeartSaVioR

**Pull request merged**